### PR TITLE
nixos/vaultwarden: update state directory name, harden systemd unit

### PR DESCRIPTION
**Pull request by mweinelt**
Align the state directory name with the module name based on the
system state version.

Harden the systemd unit and drop the capability to bind to privileged ports.

N.B.: The nixos tests have been flaky since at least 2023/10 :disappointed: 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/security/vaultwarden/default.nix [hunk 1]
```diff
@@ -176,24 +178,53 @@ in {
...
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectProc = "noaccess";
         ProtectSystem = "strict";
-        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
```
Comment (SuperSandro2000) [inline nixos/modules/services/security/vaultwarden/default.nix:185]: We need to keep this at least when the port is < 1024

Comment (mweinelt) [reply nixos/modules/services/security/vaultwarden/default.nix:185]: Why would anyone need to bind it to a privileged port though? I don't think that is a reasonable use case.

Comment (SuperSandro2000) [reply nixos/modules/services/security/vaultwarden/default.nix:185]: If you think that a not big enough group is doing that right now, then sure, just wanted to point out that this then no longer works.

Comment (mweinelt) [reply nixos/modules/services/security/vaultwarden/default.nix:185]: I'm pointing that out in the commit message, and will likely add release notes.

**Comment by SuperSandro2000** [comment]
The flaky test should be fixed with #292857. I want to merge that PR first because rebasing this one on top will be easier.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**